specs/core/1_shard-data-chains.md

```diff
@@ -359,203 +300,214 @@ def get_default_shard_state(beacon_state: BeaconState, shard: Shard) -> ShardSta
     return ShardState(
         basefee=1,
         shard=shard,
-        slot=PHASE_1_FORK_SLOT,
+        slot=ShardSlot(PHASE_1_FORK_EPOCH * SHARD_SLOTS_PER_EPOCH),
         earlier_committee_rewards=[REWARD_COEFFICIENT_BASE for _ in range(len(earlier_committee))],
         later_committee_rewards=[REWARD_COEFFICIENT_BASE for _ in range(len(later_committee))],
         earlier_committee_fees=[Gwei(0) for _ in range(len(earlier_committee))],
         later_committee_fees=[Gwei(0) for _ in range(len(later_committee))],
     )
 ```
 
-## Object validity
-
-### Shard block validation: preliminary
-
-Accept a shard block `block` only if all of the following are correct:
+#### `get_shard_base_reward`
 
-* Either `block.core.parent_root == Hash()` or a block `parent` such that `hash_tree_root(parent.core) == block.core.parent_root` has already been accepted.
-* `block.core.beacon_chain_root == get_block_root(head_beacon_state, compute_epoch_of_shard_slot(parent.core.slot))` where `head_beacon_state` is the current beacon chain head state. Alternatively phrased, a beacon chain block `beacon_ref` such that `signing_root(beacon_ref) == block.core.beacon_chain_root` has already been accepted and is part of the canonical chain, and no block with slot `beacon_ref.slot < slot <= compute_start_slot_of_epoch(compute_epoch_of_shard_slot(parent.core.slot))` is part of the canonical chain.
-* Let `beacon_state` be the state where `beacon_ref.state_root == hash_tree_root(beacon_state)`. Let `prev_state` be the post-state of the `parent` if the `parent` exists, otherwise let it be `get_default_shard_state(beacon_state, shard)` (defined below). `block.core.state_root` must equal the `hash_tree_root` of the state after applying `shard_state_transition(prev_state, beacon_state, block)`.
+```python
+def get_shard_base_reward(beacon_state: BeaconState) -> Gwei:
+    total_balance_root = integer_squareroot(get_total_active_balance(beacon_state))
+    return Gwei(REWARD_COEFFICIENT_BASE * BASE_REWARD_FACTOR // total_balance_root // BASE_REWARDS_PER_EPOCH)
+```
 
-Note that these acceptance conditions depend on the canonical beacon chain; when the canonical beacon chain reorganizes, the eligibility of shard blocks should be re-evaluated.
+### State mutators
 
-### Shard state transition function helpers
+#### `add_reward`
 
 ```python
-def add_reward(state: ShardState, beacon_state: BeaconState, index: ValidatorIndex, delta: int) -> None:
+def add_reward(state: ShardState, beacon_state: BeaconState, index: ValidatorIndex, delta: Gwei) -> None:
```

Comment:
As [`state.earlier_committee_rewards` is not in Gwei](https://github.com/ethereum/eth2.0-specs/pull/1326#discussion_r309492535), `delta` and `base_reward` are also not in Gwei.